2023/incoming/elves-versus-typos.pod

```diff
@@ -0,0 +1,175 @@
+        |
+    114 |         # the middle / hightlight, and then right of / after highlight)
+        |                        ^^^^^^^^^^
+        |
+    error: `assing` should be `assign`
+      --> 2019/articles/2019-12-14.pod:121:27
+        |
+    121 |         # what we want to assing to $x so we start drawing there again.
+
+
+There are actually a few more typos, but this is a good demonstration of what
+we might find. Now, how do we fix it?
+
+  $ typos -w 2019
+
+    error: `fils` should be `fills`, `files`, `file`
+      --> 2019/articles/2019-12-20.pod:136:59
+        |
+    136 | if that machine can access the internet, if I can put the fils in my home
+        |                                                           ^^^^
+        |
+    error: `whe` should be `when`, `we`
+      --> 2019/articles/2019-12-19.pod:25:17
+       |
+    25 | between runs.", whe Wise Old Elf explained sagely, "Then you'd be able to
+       |                 ^^^
+       |
+
+This actually fixes a number of typos, but there are two which the tool cannot
+resolve, so it makes some suggestions for us to use in a manual fix.
+
+This leaves the matter of how many changes were actually made. Let's see:
+
+    $ git diff --stat 2019
+     2019/articles/2010-12-11.pod | 2 +-
+     2019/articles/2019-12-04.pod | 2 +-
+     2019/articles/2019-12-05.pod | 4 ++--
+     2019/articles/2019-12-07.pod | 4 ++--
+     2019/articles/2019-12-10.pod | 2 +-
+     2019/articles/2019-12-12.pod | 2 +-
+     2019/articles/2019-12-13.pod | 8 ++++----
+     2019/articles/2019-12-14.pod | 6 +++---
+     2019/articles/2019-12-16.pod | 2 +-
+     2019/articles/2019-12-18.pod | 6 +++---
+     2019/articles/2019-12-20.pod | 4 ++--
+     2019/articles/2019-12-21.pod | 4 ++--
+     2019/articles/2019-12-22.pod | 2 +-
+     2019/articles/2019-12-24.pod | 2 +-
+     14 files changed, 25 insertions(+), 25 deletions(-)
+
+We can now go over the changes individually via C<git add -p> and we'll find
+that all but one seem to be correct. 24 typos found and fixed in 47 ms. That's
+not bad at all.
+
+As we process more and more files, we will like find some false positives which
+we want to correct. This can be done in a config file. Ours is C<typos.toml>.
+
+    [default.extend-words]
+    ANDed = "ANDed"
+    ba = "ba"
+    Claus = "Claus"
+    DNE = "DNE"
+    IFF = "IFF"
+    Signes = "Signes"
+
+    [files]
+    extend-exclude = ["advent.ini", "data.en"]
+
+In order to skip false positives, we add words to the C<default.extend-words>
+section. Yes, it's kind of a surprising way to do this, but let's go along with
+it. Also, any files which we don't want to check can be added to
+C<extend-include> in the C<[files]> section.
+
+=head2 More Than Just Documentation
+
+The neat thing about C<typos> is that it can also fix things in your Perl code
+itself and not just the documentation.
+
+Imagine we have a Perl script with the following contents:
+
+    #!vim perl
```

Comment:
Is this line right? I've never seen that before. Wouldn't it execute vim rather than perl?

Reply:
It will render like this.

<img width="924" alt="Bildschirmfoto 2023-12-20 um 9 58 44 AM" src="https://github.com/perladvent/Perl-Advent/assets/96205/560f43c8-596e-41ed-b6f1-f95138b0b9d4">

That appears to be the way to invoke https://metacpan.org/pod/Text::VimColor

Looks like we have the following uses in other articles:

```
rg '!vim' --no-filename --no-line-number --no-column | sed 's/^ *//g' | sort | uniq
#!vim c
#!vim css
#!vim dosini
#!vim gitcommit
#!vim golang
#!vim html
#!vim javascript
#!vim json
#!vim perl
#!vim sh
#!vim sql
#!vim text
#!vim xml
#!vim yaml
```